bin/hobby-ci.py

```diff
@@ -0,0 +1,167 @@
+name = f"do-ci-hobby-deploy-{random_bit}"
+region = 'sfo3'
+image = 'ubuntu-22-04-x64'
+size = 's-4vcpu-8gb'
+release_tag = 'latest-release'
+branch_regex = re.compile('release-*.*')
+branch = sys.argv[1]
+if branch_regex.match(branch):
+	release_tag = f"{branch}-unstable"
+hostname = f'{name}.posthog.cc'
+user_data = f'#!/bin/bash \n' \
+			f'wget https://raw.githubusercontent.com/posthog/posthog/HEAD/bin/deploy-hobby \n' \
+			f'chmod +x deploy-hobby \n' \
+			f'./deploy-hobby {release_tag} {hostname} 1 \n'
+token = os.getenv("DIGITALOCEAN_TOKEN")
+
+# Placeholders for DO resources
+domain = None
+droplet = None
+entry = None
+
+
+def block_until_droplet_is_started(droplet):
+	actions = droplet.get_actions()
+	up = False
+	while not up:
+		for action in actions:
+			action.load()
+			if action.status == 'completed':
+				up = True
+				print(action.status)
+			else:
+				print("Droplet not booted yet - waiting a bit")
+				time.sleep(5)
+
+
+def get_public_ip(droplet):
+	ip = None
+	while not ip:
+		time.sleep(1)
+		droplet.load()
+		ip = droplet.ip_address
+	print(f"Public IP found: {ip}")
+	return ip
+
+
+def create_droplet(ssh_enabled=False):
+	keys = None
+	if ssh_enabled:
+		manager = digitalocean.Manager(token=token)
+		keys = manager.get_all_sshkeys()
+	droplet = digitalocean.Droplet(
+		token=token,
+		name=name,
+		region=region,
+		image=image,
+		size_slug=size,
+		user_data=user_data,
+		ssh_keys=keys,
+		tags=['ci']
+	)
+	droplet.create()
+	return droplet
+
+
+def waitForInstance(hostname, timeout=20, retry_interval=15):
+	# timeout in minutes
+	# return true if success or false if failure
+	print("Attempting to reach the instance")
+	print(f"We will time out after {timeout} minutes")
+	url = f"https://{hostname}/_health"
+	timeout_seconds = timeout * 60
+	start_time = datetime.datetime.now()
+	while True:
+		try:
+			r = requests.get(url, verify=False)
+		except Exception as e:
+			print(f"Host is probably not up. Received exception\n{e}")
+			time.sleep(retry_interval)
+			continue
+		elapsed = datetime.datetime.now() - start_time
+		if r.status_code == 200:
+			print("Success - received heartbeat from the instance")
+			return True
+		if elapsed.seconds >= timeout_seconds:
+			print("Failure - we timed out before receiving a heartbeat")
+			return False
+		print("Instance not ready - sleeping")
+		time.sleep(retry_interval)
+
+
+def destroy_environment(droplet, domain, record, retries=3):
+	print("Destroying the droplet")
+	attempts = 0
+	while attempts <= retries:
+		attempts += 1
+		try:
+			droplet.destroy()
+			break
+		except Exception as e:
+			print(f"Could not destroy droplet because\n{e}")
+	print("Destroying the DNS entry")
+	attempts = 0
+	while attempts <= retries:
+		attempts += 1
+		try:
+			domain.delete_domain_record(id=record['domain_record']['id'])
+			break
+		except Exception as e:
+			print(f"Could not destroy the dns entry because\n{e}")
+
+
+def handle_sigint():
+	global droplet
```

Comment:
In general I think we should discourage the use of global variables

Reply:
Totally agree. I was approaching this as a script vs app, but in principal 100% agree.